beefy-node/node/src/service.rs

```diff
@@ -248,21 +248,21 @@ pub fn new_full(mut config: Configuration) -> Result<TaskManager, ServiceError>
 		None
 	};
 
-	let beefy_params = beefy_gadget::BeefyParams {
+	let beefy_params = beefy_gadget::BeefyParams::<_, beefy_primitives::ecdsa::Pair, _, _, _> {
 		client,
 		backend,
-		key_store: keystore.clone(),
+		// TODO [ToDr]  this is shit we have an Arc<Arc<>> here, I'm sure we can do better.
+		key_store: keystore.as_ref().map(|x| Arc::new(x.clone()) as _),
```

Comment:
Maybe we can add a helper method at least so that this nightmare is not replicated in every downstream service.
Something like `BeefyParams::cast_keystore(keystore)`